src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -101,6 +122,91 @@ public void resetOnPath(String path) {
         // Intentionally blank
     }
 
+    /**
+     * Helper method which handles the file creation for both normal file uploads and for markdown
+     *
+     * @param file         The file to be uploaded with a maximum file size set in resources/config/application.yml
+     * @param keepFileName specifies if original file name should be kept
+     * @param markdown     boolean which is set to true, when we are uploading a file within the markdown editor
+     * @return The path of the file
+     */
+    @NotNull
+    public String handleSaveFile(MultipartFile file, boolean keepFileName, boolean markdown) {
+        // check for file type
+        String filename = file.getOriginalFilename();
+        if (filename == null) {
+            throw new IllegalArgumentException("Filename cannot be null");
+        }
+        // sanitize the filename and replace all invalid characters with "_"
+        filename = filename.replaceAll("[^a-zA-Z\\d\\.\\-]", "_");
+        String fileExtension = FilenameUtils.getExtension(filename);
+        if (this.allowedFileExtensions.stream().noneMatch(fileExtension::equalsIgnoreCase)) {
```

Comment:
```suggestion
        if (!this.allowedFileExtensions.contains(fileExtension.toLowerCase())) {
```
This seams more intuitive to me